src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -453,7 +451,7 @@ static IEnumerable<NetworkAddress> Distinct(IEnumerable<NetworkAddress> aa, ISet
                 yield return address;
 
             // get configured addresses that haven't been tried already
-            addresses = _addressProvider.GetAddresses();
+            addresses = _clusterState.AddressProvider.GetAddresses();
```

Comment:
I didn't get why `AddressProvider` is moved to `ClusterState`.

Reply:
Because it's now used both by `ClusterConnections` and `ClusterMembers` and at one point it seemed clever. Now, `ClusterConnections` has a reference to `ClusterMembers` so we *could* have `AddressProvider` owned by `ClusterMembers` instead. I am just not too sure. What do you think?

Reply:
Yeah, it's clever.